### ISSUE TITLE
Sesh close command

### ISSUE THREAD

**Issue by joshmedeski** [joshmedeski/sesh#166]
### What would you like sesh to do?

`sesh close` to improve the functionality of tmux

- When closing the last window of a session, we want to close and connect to the last attached session (tmux currently chooses something apparently random?)
- **Alternative solution:** Look into contributing to tmux source code and fixing this issue (see https://github.com/tmux/tmux/issues/3343) (also see this: https://github.com/tmux/tmux/blob/37771a5a8de65d7ddfef76d26810fe521ead0a77/server-fn.c?plain=1#L432)

**Comment by ajkavanagh** [comment]
Yes, I ran into this. It turns out that tmux has a nice feature which is "detach-on-destroy" that can be set to "no-detached". I have this in my `tmux.conf` and the behaviour is that when I close the last window of a session it reverts to the previous session rather than exiting out of the client and requiring a `tmux a` to get back in:

```
# Client detaches only if there are no detached sessions;
# if detached sessions exist, the client is switched to the most recently active.
set-option -g detach-on-destroy no-detached
```

Hope this is useful to someone.

**Comment by joshmedeski** [comment]
I was able to use tmux's hook feature to trigger a cache rebuild.

```
set-hook -g session-closed  'run-shell -b "sesh cache refresh"'
```

I probably can't use this to try and fix the attach to last session with this mechanism. Perhaps I need to make a contribution directly to tmux.

**Comment by ajkavanagh** [comment]
HI @joshmedeski ; is it necessary to do a `sesh cache refresh` after closing the session?  I'm also wondering if using both of them might work:

```
set-option -g detach-on-destroy no-detached
set-hook -g session-closed  'run-shell -b "sesh cache refresh"'
```

i.e. default behaviour is to not exit tmux and also to run the `sesh cache refresh` command.  (this may have been what you meant, in hindsight!).
 
One thing I'm not sure about is how a stale cache would present itself?

**Comment by joshmedeski** [comment]
When you close a tmux session and open sesh quickly, the closed session will still appear as a tmux session. I also disabled detach on destroy but sometimes I'm moving around quickly between sessions after closing one.

Hope that makes sense.

**Comment by joshmedeski** [comment]
BTW you have to opt-into the cache feature, so if you're not using it then no need to add the hook.

**Comment by joshmedeski** [comment]
I'm gonna stick to the detach-on-destroy behavior, it does a good job and has a few decent options.
